### DISCUSSION TITLE
Pattern rule not working in React Native

### DISCUSSION THREAD

**Discussion by KyleAsaff** [react-hook-form/react-hook-form#3791] [Q&A]
I have the following code:

```
...
<Item stackedLabel error={!!errors.userName}>
  <Label>Username*</Label>
  <Controller
    name="userName"
    control={control}
    defaultValue=""
    rules={{
      required: 'test',
      pattern: {
        value: /[A-Za-z]{3}/,
        message: 'error message',
      },
    }}
    render={({ onChange, onBlur, value }) => (
      <Input placeholder="" ref={usernameRef} />
    )}
  />
</Item>
...
```

`required` will throw an error when the input is blank but `pattern` is not throwing an error (when typing something like '######' in the input).

Is there anything I am doing wrong in my syntax? Even when I try without `required` the `pattern` rule won't. I am following the docs exactly.

version:
```
    "react-hook-form": "^6.13.1",
    "react-native": "https://github.com/expo/react-native/archive/sdk-40.0.1.tar.gz",
```

**Comment by bluebill1049** [comment]
i would try to reproduce that in a codesandbox first, it should worked the same.
